### PR TITLE
fix: make sure router gets built on start, for #5040

### PR DESCRIPTION
**Pull request by rfay**
## The Issue

* https://github.com/ddev/ddev/issues/5040

We've been having trouble on upgrades with ddev-router not getting rebuilt; it turns out the docker-compose command didn't have --build in there. 

## How This PR Solves The Issue

Add `--build` so router gets rebuilt if it already existed.

## Release/Deployment Notes

<a href="https://gitpod.io/#https://github.com/ddev/ddev/pull/5052"></a>

### CODE REVIEW

**Comment by rfay** [comment]
This solved @rpkoller 's problem, this alone, was tested on misbehaving system.

**Comment by rpkoller** [comment]
I hope that is the correct bit @rfay asked for. but that is the output we had debugging the problem on my computer.

in `~/.ddev/router-build` we did `cat Dockerfile`

```
#ddev-generated

ARG BASE_IMAGE
FROM $BASE_IMAGE


RUN apk add bash curl htop vim
WORKDIR /mnt/ddev-global-cache/traefik
COPY /traefik_healthcheck.sh /healthcheck.sh
```

and then the following command inspecting the router output: 
```
$> docker inspect --format "{{json .State.Health }}" ddev-router
{"Status":"unhealthy","FailingStreak":1003,"Log":[{"Start":"2023-06-30T21:49:05.202976174Z","End":"2023-06-30T21:49:05.321276965Z","ExitCode":127,"Output":"/bin/sh: /healthcheck.sh: not found\n"},{"Start":"2023-06-30T21:49:06.328373341Z","End":"2023-06-30T21:49:06.445561549Z","ExitCode":127,"Output":"/bin/sh: /healthcheck.sh: not found\n"},{"Start":"2023-06-30T21:49:07.450499216Z","End":"2023-06-30T21:49:07.576819591Z","ExitCode":127,"Output":"/bin/sh: /healthcheck.sh: not found\n"},{"Start":"2023-06-30T21:49:08.582849508Z","End":"2023-06-30T21:49:08.688931134Z","ExitCode":127,"Output":"/bin/sh: /healthcheck.sh: not found\n"},{"Start":"2023-06-30T21:49:09.694490134Z","End":"2023-06-30T21:49:09.799587759Z","ExitCode":127,"Output":"/bin/sh: /healthcheck.sh: not found\n"}]}```

If i should anything else please let me know. and again thanks for that  quick fix and the spontaneous debugging session!

File: pkg/ddevapp/router.go [hunk 1]
```diff
@@ -102,7 +102,7 @@ func StartDdevRouter() error {
 	}
 
 	// run docker-compose up -d against the ddev-router full compose file
-	_, _, err = dockerutil.ComposeCmd([]string{routerComposeFullPath}, "-p", RouterProjectName, "up", "-d")
+	_, _, err = dockerutil.ComposeCmd([]string{routerComposeFullPath}, "-p", RouterProjectName, "up", "--build", "-d")
```
Comment (gilbertsoft) [inline pkg/ddevapp/router.go:105]: Wondering if we also should use `--force-recreate` like for the SSH agent container or would this break something?

**Review by gilbertsoft — APPROVED** [review]

**Comment by gilbertsoft** [comment]
@rpkoller can you please check if the binaries from https://github.com/ddev/ddev/pull/5052#issuecomment-1615237195 work for you? Thanks!

**Review by gilbertsoft — APPROVED** [review]

**Comment by rfay** [comment]
@rpkoller had already tested earlier version; the complexity is once the problem is solved (once the new image gets built) the problem is solved. So he can't test without going back to earlier version (with traefik enabled), deleting all build cache, deleting -built image, and then upgrading.

**Comment by rfay** [comment]
Thanks for fixing in https://github.com/ddev/ddev/pull/5052/commits/8f409c7c3cafe8a85f733b0fe8b135228444422e and the refactor of the embeds is 💯

**Pull request merged**